marklogic-data-hub/src/test/java/com/marklogic/hub/hubcentral/migration/AllStepsReferencedTest.java

```diff
@@ -0,0 +1,429 @@
+
+class AllStepsReferencedTest extends AbstractHubCoreTest {
+
+    ObjectMapper mapper = new ObjectMapper();
+
+    @Autowired
+    FlowRunner flowRunner;
+
+    @BeforeEach
+    void setUp() {
+        String folderInClasspath = "test-projects/all-steps-referenced";
+        installProjectInFolder(folderInClasspath);
+
+        // copy input directory to ye-olde-project
+        try {
+            File testProjectDir = new ClassPathResource(folderInClasspath).getFile();
+            HubProject hubProject = getHubConfig().getHubProject();
+            File inputDir = new File(testProjectDir, "input");
+            File copyToDir = new File(hubProject.getProjectDir().toFile(), "input");
+            if (inputDir.exists()) {
+                FileUtils.copyDirectory(inputDir, copyToDir);
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Unable to load project files: " + e.getMessage(), e);
+        }
+    }
+
+    @Test
+    void validateMigratedFlowsAndSteps() {
+        HubConfig hubConfig = getHubConfig();
+
+        FlowMigrator flowMigrator = new FlowMigrator(hubConfig);
+        flowMigrator.migrateFlows();
+
+        validateFullyReferencedFlowsAndSteps();
+    }
+
+    @Test
+    void migrateAndRunCompleteFlow() {
+        HubConfig hubConfig = getHubConfig();
+
+        FlowMigrator flowMigrator = new FlowMigrator(hubConfig);
+        flowMigrator.migrateFlows();
+        installUserArtifacts();
+
+        RunFlowResponse flowResponse = flowRunner.runFlow(new FlowInputs("CurateCustomerJSON", "1", "2", "3", "4"));
+        flowRunner.awaitCompletion();
+        RunStepResponse runStepResponse = flowResponse.getStepResponses().get("1");
+        assertTrue(runStepResponse.isSuccess(), "Ingestion step failed!");
+        runStepResponse = flowResponse.getStepResponses().get("2");
+        assertTrue(runStepResponse.isSuccess(), "Mapping step failed!");
+        runStepResponse = flowResponse.getStepResponses().get("3");
+        assertTrue(runStepResponse.isSuccess(), "Matching step failed!");
+        runStepResponse = flowResponse.getStepResponses().get("4");
+        assertTrue(runStepResponse.isSuccess(), "Merging step failed!");
+
+        assertTrue(getFinalDocCount("merged-customer") == 6,"Expected 6 merged-customer docs");
+    }
+
+    void migrateAndRunMasteringFlow() {
+        HubConfig hubConfig = getHubConfig();
+
+        FlowMigrator flowMigrator = new FlowMigrator(hubConfig);
+        flowMigrator.migrateFlows();
+        installUserArtifacts();
+
+        // The following test was copied from MasterTest
+
+        RunFlowResponse flowResponse = flowRunner.runFlow(new FlowInputs("myNewFlow", "1","2","3"));
+        flowRunner.awaitCompletion();
+        RunStepResponse masterJob = flowResponse.getStepResponses().get("3");
+        assertTrue(masterJob.isSuccess(), "Mastering job failed!");
+        assertTrue(getFinalDocCount("sm-person-merged") >= 10,"At least 10 merges occur");
```

Comment:
I think you should be able to run the mastering test on the same data that you used for the matching and merging test, which I highly recommend because we know there's exactly one match expected. We don't want to repeat the assertions from MasterTest, where we expect a very high count of documents, and when that count isn't met, it's nearly impossible to determine why not.

Reply:
Will do